src/Types.hs

```diff
@@ -551,20 +579,20 @@ type instance IdOf User             = AUID User
 type instance IdOf Idea             = AUID Idea
 type instance IdOf Topic            = AUID Topic
 type instance IdOf Delegation       = AUID Delegation
-type instance IdOf Comment          = AUID Comment
--- ^ A more precise identifier would be:
---      IdOf Comment = (AUID Idea, NonEmpty (AUID Comment))
--- This would be the full path from AulaData down to the comment
--- which could be a reply to reply...
-type instance IdOf CommentVote      = AUID User
-type instance IdOf IdeaVote         = AUID User
-type instance IdOf IdeaLike         = AUID User
--- ^ If we were to have more precise comment identifiers these 3 could
--- become: (IdOf Comment, AUID User)
+type instance IdOf Comment          = CommentId
+type instance IdOf CommentVote      = CommentVoteId
+type instance IdOf IdeaVote         = IdeaVoteLikeId
+type instance IdOf IdeaLike         = IdeaVoteLikeId
 type instance IdOf IdeaVoteResult   = AUID IdeaVoteResult
 type instance IdOf IdeaJuryResult   = AUID IdeaJuryResult
 
-type AMap a = Map (IdOf a) a
+type family   AUIDOf a
```

Comment:
The A stands for Aula. The status of these keys/ids has to be discussed since these types both evolved too far and far enough such that we start to see how to clean and improve. The distinction I want to make is between what serves as the key for final map in which the record is located and the parent keys needed to traverse the whole structure from the top.

I think that the first kind of key can be set to always be an `AUID` and the type of parent keys can be driven with a type family similar to what we have now.